File: src/main/java/com/frightanic/smn/api/Station.java
package com.frightanic.smn.api;

import com.wordnik.swagger.annotations.ApiModel;
import com.wordnik.swagger.annotations.ApiModelProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * Representation of an SMN observation station.
 */
@Getter
@AllArgsConstructor
@ApiModel(value = "SMN station meta data")
public class Station {
  @ApiModelProperty(value = "3-char all upper-case station code")
  private final String code;
  @ApiModelProperty(value = "Original name in local language")
  private final String name;
  @ApiModelProperty(value = "CH1903 (Swiss grid) y-axis value")
  private final int ch1903Y;
  @ApiModelProperty(value = "CH1903 (Swiss grid) x-axis value")
  private final int ch1903X;
  @ApiModelProperty(value = "WGS84 latitude")
  private final double lat;
  @ApiModelProperty(value = "WGS84 longitude")
  private final double lng;
  @ApiModelProperty(value = "meters above sea level")
  private final int elevation;
}

File: src/main/kotlin/com/frightanic/smn/api/Station.kt
package com.frightanic.smn.api

import org.eclipse.microprofile.openapi.annotations.media.Schema

@Schema(description = "SMN station meta data")
data class Station(
    @param:Schema(description = "3-char all upper-case station code", required = true, readOnly = true)
    val code: String,

    @param:Schema(description = "Original name in local language", required = true, readOnly = true)
    val name: String,

    @param:Schema(description = "CH1903 (Swiss grid) y-axis value", required = true, readOnly = true)
    val ch1903Y: Int,

    @param:Schema(description = "CH1903 (Swiss grid) x-axis value", required = true, readOnly = true)
    val ch1903X: Int,

    @param:Schema(description = "WGS84 latitude", required = true, readOnly = true)
    val lat: Double,

    @param:Schema(description = "WGS84 longitude", required = true, readOnly = true)
    val lng: Double,

    @param:Schema(description = "meters above sea level", required = true, readOnly = true)
    val elevation: Int,
)
